### PR TITLE
[DOC,ENH] `base` docs and testing

### PR DESCRIPTION
**Pull request by MatthewMiddlehurst**
Testing and documentation for base classes. Refactors a few usages of `get_class_tag` where it is unnecessary and a couple of other minor refactors which explode the file change count.

### CODE REVIEW

File: aeon/base/_base.py [hunk 1]
```diff
@@ -221,7 +226,7 @@ def get_tags(self):
         collected_tags.update(self._tags_dynamic)
         return deepcopy(collected_tags)
 
-    def get_tag(self, tag_name, tag_value_default=None, raise_error=True):
+    def get_tag(self, tag_name, raise_error=True, tag_value_default=None):
```
Comment (TonyBagnall) [inline aeon/base/_base.py:229]: these are the default arguments and not needed

File: aeon/testing/mock_estimators/_mock_classifiers.py [hunk 2]
```diff
@@ -41,26 +43,27 @@ class MockClassifierFullTags(MockClassifierPredictProba):
...
+    """Mock classifier for testing fit/predict with multiple parameters.
 
     Parameters
     ----------
     return_ones : bool, default=False
         If True, predict ones, else zeros.
     """
 
-    def __init__(self, return_ones=False):
+    def __init__(self, return_ones=False, value=50):
```
Comment (TonyBagnall) [inline aeon/testing/mock_estimators/_mock_classifiers.py:55]: why does it have a parameter value now?

**Review by TonyBagnall — APPROVED** [review]
LGTM

File: aeon/testing/mock_estimators/_mock_classifiers.py (hunk 2 above)
Comment (MatthewMiddlehurst) [reply aeon/testing/mock_estimators/_mock_classifiers.py:55]: just a useless param to help with testing

File: aeon/base/_base.py (hunk 1 above)
Comment (MatthewMiddlehurst) [reply aeon/base/_base.py:229]: little confused, are there calls to the method which input the default?

Comment (TonyBagnall) [reply aeon/base/_base.py:229]: yes comment in wrong place! There was one call with explicit default parameters, but it doenst matter

**Pull request merged**